src/Application/Composer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application;
+
+use Composer\Semver\Comparator;
+use Composer\Semver\Constraint\Constraint;
+use Composer\Semver\Constraint\MultiConstraint;
+use Composer\Semver\Semver;
+use Composer\Semver\VersionParser;
+use Illuminate\Support\Str;
+
+/**
+ * @internal
+ */
+final class Composer
+{
+    /** @var array<string, string|int|array> */
+    private $config;
+
+    /**
+     * Composer constructor.
+     *
+     * @param array<string, string|int|array> $data
+     */
+    public function __construct(array $data)
+    {
+        $this->config = $data;
+    }
+
+    public static function fromPath(string $path): self
+    {
+        return new self(json_decode((string) file_get_contents($path), true));
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public function getRequirements(): array
+    {
+        return $this->config['require'] ?? [];
+    }
+
+    /**
+     * @return array<string, string>
+     */
+    public function getReplacements(): array
+    {
+        return $this->config['replace'] ?? [];
+    }
+
+    public function getName(): ?string
+    {
+        return $this->config['name'] ?? null;
+    }
+
+    public function getPhpVersion(): ?string
+    {
+        return $this->getRequirements()['php'] ?? null;
+    }
+
+    public function lowestPhpVersionIsGreaterThenOrEqualTo(string $version): ?bool
+    {
+        $composerVersion = $this->getPhpVersion();
+
+        if ($composerVersion === null) {
+            return null;
+        }
+        $composerVersion = Str::after($composerVersion, '^');
```

Comment:
Be careful. Illuminate/support is only in `require-dev`.

Reply:
Oh yeah, great catch, we can easily change this, just used to using it :)